gammapy/detect/cwt.py

```diff
@@ -434,115 +412,195 @@ class CWTData(object):
+
+    @property
+    def approx_bkg(self):
+        """
+        2D `~gammapy.image.SkyImage` image. In the course of iterations
+        updated by convolution of ``background`` with ``kern_approx``.
+        Primordial initialized by zero array.
+        """
+        return SkyImage(name='approx_bkg', data=self._approx_bkg, wcs=self._wcs)
+
+    @property
+    def transform_2d(self):
+        """
+        2D `~gammapy.image.SkyImage` image. Created from transform_3d by summarize
+        values per 0 axes. Primordial initialized by zero array.
+        """
+        return SkyImage(name='transform_2d', data=self._transform_2d, wcs=self._wcs)
+
+    @property
+    def support_2d(self):
+        """
+        2D '~gammapy.cube.SkyCube' cube exclusion mask. Created from support_3d
+        by OR-operation per 0 axis.
+        """
+        support_2d = (self._support.sum(0) > 0)
+        return SkyImage(name='support_2d', data=support_2d, wcs=self._wcs)
+
+    @property
+    def residual(self):
+        """
+        2D `~gammapy.image.SkyImage` image. Calculate as
+        ``counts - model - approx``.
+        """
+        residual = self._counts - (self._model + self._approx)
+        return SkyImage(name='residual', data=residual, wcs=self._wcs)
+
+    @property
+    def model_plus_approx(self):
+        """TODO: document what this is."""
+        return SkyImage(name='/model plus approx',
+                        data=self._model + self._approx,
+                        wcs=self._wcs)
+
+    @property
+    def transform_3d(self):
+        """
+        3D `~gammapy.cube.SkyCube` cube. Primordial initialized by zero array. In the course of
+        iterations updated by convolution of ``counts - total_background`` with kernel
+        for each scale (``total_background = model + background + approx``).
+        """
+        return SkyCube(name='transform_3d', data=self._transform_3d, wcs=self._wcs)
+
+    @property
+    def error(self):
+        """
+        3D `~gammapy.cube.SkyCube` cube. Primordial initialized by zero array.
+        In the course of iterations updated by convolution of ``total_background``
+        with kernel^2 for each scale.
+        """
+        return SkyCube(name='error', data=self._error, wcs=self._wcs)
+
+    @property
+    def support_3d(self):
+        """
+        3D '~gammapy.cube.SkyCube' cube exclusion mask. Primordial initialized by zero array.
+        """
+        return SkyImage(name='support_3d', data=self._support, wcs=self._wcs)
 
-        # Strange result after each iteration of CWT...
-        self.mysterious_result = np.zeros(shape_2d)
+    @property
+    def max_scale_image(self):
+        """Compute the maximum scale image as '~gammapy.image.SkyImage'."""
+        # idx_scale_max = np.argmax(self._transform_3d, axis=0)
+        # return kernels.scales[idx_scale_max] * (self._support.sum(0) > 0)
+        transform_2d_max = np.max(self._transform_3d, axis=0)
```

Comment:
I don't create property for this image, because we already have transform_2d, but this image we calculate by another way. We can make a parameter and remove property from `CWTData.transform_2d(mode='max/sum')`. @cdeil what do you think?